rfc/rfc-59/rfc-59.md

```diff
@@ -0,0 +1,285 @@
+  * returns true of false, means whether persisted record could be updated.
+  */
+  protected boolean needUpdatingPersistedRecordWithMultipleFieldsVerification(IndexedRecord persistedRecord, IndexedRecord incomingRecord, Properties properties) ;
+
+  // invoked by needUpdatingPersistedRecordWithMultipleFieldsVerification, to select which event-time fields are not null.
+  // Since null value usually has different meaning, it should be treated seriously so that we create a function to do this.
+  private Map<String, Comparable<?>> chooseAndGetMultipleCombineFields(List<String> MultipleEventimeFields, IndexedRecord incomingRecord) ;
+
+  // Return true if value equals defaultValue otherwise false.
+  public Boolean overwriteField(Object value, Object defaultValue) ;
+
+  protected boolean isDeleteRecord(GenericRecord genericRecord) ;
+
+  // Override methods in order to implement the interface HoodieRecordPayload
+  // Compatible with current workflow
+  @Override
+  public VerifyMultipleLatestEventTimesPayload preCombine(VerifyMultipleLatestEventTimesPayload oldValue) ;
+
+  @Override
+  // IF AND ONLY IF all event-time fields are same or newer can Hudi updates all notNull value from incomingRecord to persistedReocrd
+  public Option<IndexedRecord> combineAndGetUpdateValue(IndexedRecord currentValue, Schema schema, Properties properties) throws IOException ;
+
+  @Override
+  public Option<IndexedRecord> getInsertValue(Schema schema) throws IOException ;
+
+  @Override
+  public Option<Map<String, String>> getMetadata() ;
+}
+
+```
+
+### Deduplicate record with multiple event-time fields
+
+This function will deduplicate records with same primay key(s) by multiple event-time fields and only keep record with latest values in every event-time field.
+Since Hudi buffers records in a buckect before flushing or checkpointing, so it's very necessary to conduct deduplication inside a bucket in order to sink only one record (whose every event-time is latest) to next step where this record will compare with record on Disk to decide which one should be remained or should some part of the persisted record be updated.
+
+```java
+// hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/commit/FlinkWriteHelper.java
+public class FlinkWriteHelper<T extends HoodieRecordPayload, R> extends AbstractWriteHelper<T, List<HoodieRecord<T>>, List<HoodieKey>, List<WriteStatus>, R> {
+  
+  // ...
+
+  /**
+  *
+  * This function deduplicates records having same primary keys by multiple event-time fields. The only record with the latest event-time fields will be remained.
+  * It's very like another function (FlinkWriteHelper # deduplicateRecords) in this class. The difference is that this function use multiple event-time to deduplicate records.
+  */
+  public List<HoodieRecord<T>> deduplicateRecordsWithMultipleCombineFields(List<HoodieRecord<T>> records, HoodieIndex<T, ?, ?, ?> index, int parallelism) ;

```

Comment:
I don't have much context on Flink classes design and layering. but would be good to avoid adding new methods and just add overloaded methods for existing ones.